File: examples/simple/simple.js
#!/usr/bin/env gjs

imports.gi.versions.Gray = "0.1";
imports.gi.versions.Gtk = "3.0";
const { Gtk, Gray } = imports.gi;


function main() {
    Gtk.init(ARGV);

    let watcher = new Gray.Watcher();
    watcher.connect(
        "item-added",
        (self, identifier) => {
            item = watcher.get_item_for_identifier(identifier);
            print(item.get_title());
        }
    );

    Gtk.main();
}

main();

File: examples/simple/simple.py
import gi

gi.require_version("Gray", "0.1")
from gi.repository import Gray, Gtk


def main():
    watcher = Gray.Watcher()
    watcher.connect(
        "item-added",
        lambda _, identifier: print(
            watcher.get_item_for_identifier(identifier).get_title()
        ),
    )

    Gtk.main()


if __name__ == "__main__":
    main()
